src/NuGet.Core/NuGet.Packaging/RuntimeModel/RuntimeGraph.cs

```diff
@@ -245,5 +305,69 @@ public static void ReleaseList(List<T> list)
                 }
             }
         }
+
+        /// <summary>
+        /// RID + package id
+        /// </summary>
+        private sealed class RuntimeDependencyKey : IEquatable<RuntimeDependencyKey>
```

Comment:
Do we really need to create a new class here?

Reply:
I've heard Tuple is much slower for equality checks. I could use BenchMarkDotNet to try, but I think the extra class is low risk. This is also a known hot path that I feel warrants the extra code.

Reply:
Here are the results from a test that adds 2 keys to a dictionary, then does 1 lookup with success, and 1 with a miss:

```
      Method |     Mean |    Error |   StdDev |
------------ |---------:|---------:|---------:|
 HelperClass | 229.7 ns | 1.661 ns | 1.553 ns |
       Tuple | 416.7 ns | 4.300 ns | 4.022 ns |
```

Having the extra class looks faster